src/advanced_inv.cpp

```diff
@@ -361,13 +362,22 @@ void advanced_inventory::print_items( advanced_inventory_pane &pane, bool active
         mvwprintz( window, 6 + x, weight_startpos, print_color, "%5.*f", w_precision, it_weight );
 
         //print volume column
-        int it_vol = sitem.volume / units::legacy_volume_factor;
+        double it_vol = units::to_liter( sitem.volume );
+        size_t v_precision;
         print_color = ( it_vol > 0 ) ? thiscolor : thiscolordark;
-        if( it_vol > 9999 ) {
-            it_vol = 9999;
-            print_color = selected ? hilite( c_red ) : c_red;
+        if( it_vol >= 100 ) {
+            if( it_vol >= 10000 ) {
+                it_vol = 9999;
+                print_color = selected ? hilite( c_red ) : c_red;
+            }
+            v_precision = 0;
+        } else if( it_vol >= 10 ) {
```

Comment:
> My thinking behind this change is: the vol column only has 4 digits worth of space
> will make this column overflow to the left. 

This is a good point and something I hadn't considered. What you have is for now an improvement (and thanks for identifying #18430 in the process) so lets go with this for now